checks/raw/pinned_dependencies.go

```diff
@@ -498,3 +494,28 @@ var validateGitHubActionWorkflow fileparser.DoWhileTrueOnFileContent = func(
 
 	return true, nil
 }
+
+func isActionDependencyPinned(actionUses string) bool {
+	// Though GitHub Actions also allows actions to be specified as local filepaths
+	// (e.g. uses: <path/to/file/in/repo>), it's not yet practical to determine whether an
+	// arbitrary file that lives in the action context is a trustworthy, pinned workflow file
+	// So, those actions are treated as 'unpinned'
+
+	publicActionRegex := regexp.MustCompile(`^([\w-]+)/([\w-]+)(/([\w-/]+))?(@([a-f\d]+)|([\w\.-]+))?`)
+
+	if m := publicActionRegex.FindStringSubmatch(actionUses); len(m) > 0 {
+		ref := m[6]
+		// Ensure a hash at least as large as SHA1 is used (40 hex characters).
+		return len(ref) >= 40
+	}
+
+	//nolint:lll
+	dockerhubActionRegex := regexp.MustCompile(`docker://([\w\.-]+/)?([\w\.-]+)(/[\w\.-:]+)?((@sha256:([a-f\d]+))|(:[\w\.\-]+))?`)
```

Comment:
My understanding is `_` is also valid in docker image names.

e.g. https://github.com/falconry/falcon/blob/0d808d99e83bc1aba6e062c4fa4433e32cb8c3b3/.github/workflows/create-wheels.yaml#L266

Docker hosts might have colons for a port? 
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-a-docker-public-registry-action

I noticed some `:` in a later group: `(/[\w\.-:]+)?`
That might allow multiple colons, not sure.

Reply:
Changed this to `.`.